src/osx/cocoa/textctrl.mm

```diff
@@ -1082,14 +1114,39 @@ - (void)controlTextDidEndEditing:(NSNotification *)aNotification
     if ( start == -1 && end == -1 )
     {
         NSMutableDictionary* const
-            attrs = [NSMutableDictionary dictionaryWithCapacity:3];
+            attrs = [NSMutableDictionary dictionaryWithCapacity:5];
         if ( style.HasFont() )
             [attrs setValue:style.GetFont().OSXGetNSFont() forKey:NSFontAttributeName];
         if ( style.HasBackgroundColour() )
             [attrs setValue:style.GetBackgroundColour().OSXGetNSColor() forKey:NSBackgroundColorAttributeName];
         if ( style.HasTextColour() )
             [attrs setValue:style.GetTextColour().OSXGetNSColor() forKey:NSForegroundColorAttributeName];
+        if ( style.HasFontUnderlined() )
+        {
+            switch ( style.GetUnderlineType() )
+            {
+                case wxTEXT_ATTR_UNDERLINE_NONE:
+                    [attrs setObject:[NSNumber numberWithInt:( NSUnderlineStyleNone )] forKey:NSUnderlineStyleAttributeName];
```

Comment:
We should do the same thing as in wxGTK code, i.e. map our value to Cocoa one in the `switch` and then call `setObject:numberWithInt:` only once after it, it's really unnecessary to repeat this call 4 times.